arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/EffectSpec.kt

```diff
@@ -241,8 +242,27 @@ class EffectSpec :
         }.runCont()
       } shouldBe Either.Left(e)
     }
+    
+    "Dispatching in nested Effect causes incorrect behavior" {
```

Comment:
I am a bit confused by the name here; is this checking that the incorrect behavior is there, or that it's not there?

Reply:
Hmm yes, naming is not great.
I wrote the test to verify the incorrect behavior, and ran it locally and on Github CI since I wanted to actually see it fail there. https://github.com/arrow-kt/arrow/pull/2766

Then I fixed it, and left the naming there. Like I explained above it's not entirely clear what is/was actually going wrong. Everything is pointing to the Kotlin compiler getting confused, and wiring the `Continuation`s wrong. I would say this fix is a great one, since it also removes unnecessary `Continuation` allocations.

Reply:
@serras I think I found a good name for this test! Thanks for flagging it, it was indeed a very confusing name.

Reply:
Top!